execution-engine/src/wasi/wasmi.rs

```diff
@@ -386,20 +437,27 @@ impl ModuleImportResolver for WASMIRuntimeState {
     /// "Resolves" a H-call by translating from a H-call name, `field_name` to
     /// the corresponding H-call code, and dispatching appropriately.
     fn resolve_func(&self, field_name: &str, signature: &Signature) -> Result<FuncRef, Error> {
-        let index = WasiAPIName::try_from(field_name).map_err(|_| {
-            Error::Instantiation(format!(
+        let index = if let Ok(x) = WasiAPIName::try_from(field_name) {
+            APIName::WasiAPIName(x)
+        } else if let Ok(x) = VeracruzAPIName::try_from(field_name) {
+            APIName::VeracruzAPIName(x)
+        } else {
+            return Err(Error::Instantiation(format!(
                 "Unknown function {} with signature: {:?}.",
                 field_name, signature
-            ))
-        })?;
+            )));
+        };
 
         if !TypeCheck::check_signature(index, signature) {
             Err(Error::Instantiation(format!(
                 "Function {} has an unexpected type-signature: {:?}.",
                 field_name, signature
             )))
         } else {
-            Ok(FuncInstance::alloc_host(signature.clone(), index as usize))
+            Ok(FuncInstance::alloc_host(
+                signature.clone(),
+                index.to_usize().unwrap(),
```

Comment:
Can this `unwrap()` abort at runtime?

Reply:
I don't think so. Is it worth investigating?

Reply:
The default conversion functions are used for `WasiAPIName` and `VeracruzAPIName`, so presumably they return a non-negative number that can't be too big for a `usize`.